src/Form/BusinessAccountType.php

```diff
@@ -21,21 +21,26 @@
 
 class BusinessAccountType extends AbstractType
 {
-    private $authorizationChecker;
-    private $objectManager;
-
     public function __construct(
-        AuthorizationCheckerInterface $authorizationChecker,
-        EntityManagerInterface $objectManager)
+        private readonly AuthorizationCheckerInterface $authorizationChecker,
+        private readonly EntityManagerInterface $objectManager)
     {
-        $this->authorizationChecker = $authorizationChecker;
-        $this->objectManager = $objectManager;
     }
 
     public function buildForm(FormBuilderInterface $builder, array $options)
     {
         $builder
-            ->add('name', TextType::class, ['label' => 'registration.step.company.name'])
+            ->add('name', TextType::class, [
+                'label' => 'registration.step.company.name'
+            ])
+            ->add('legalName', TextType::class, [
+                'label' => 'registration.step.company.legal_name',
+                'required' => !$this->authorizationChecker->isGranted('ROLE_ADMIN'),
```

Comment:
Maybe they should be hidden for non-admin, no ?

Reply:
No, they are for `business manager` to setup. So an admin can skip them while creating a business account, but a `business manager` must enter them